sdk/attestation/Azure.Security.Attestation/src/Models/AttestationSigner.cs

```diff
@@ -19,9 +19,9 @@ public class AttestationSigner
         /// </summary>
         /// <param name="signingCertificates"></param>
         /// <param name="certificateKeyId"></param>
-        public AttestationSigner(X509Certificate2[] signingCertificates, string certificateKeyId)
+        public AttestationSigner(IEnumerable<X509Certificate2> signingCertificates, string certificateKeyId)
         {
-            SigningCertificates = signingCertificates;
+            SigningCertificates = (IReadOnlyList<X509Certificate2>)signingCertificates;
```

Comment:
```suggestion
            SigningCertificates = signingCertificates switch =>
            {
                IReadOnlyList<X509Certificate2> certificateList => certificateList,
                _ => signingCertificates.ToList().AsReadOnly()
            };
```

Since the enumerable is caller-provided, there's no guarantee that it's safe to cast.  This allows us to test the cast and avoid allocating if we already have a read-only list, and safely do the conversion if we do not.